packages/opentelemetry-tracing/src/export/BatchSpanProcessor.ts

```diff
@@ -109,8 +109,12 @@ export class BatchSpanProcessor implements SpanProcessor {
     return new Promise((resolve, reject) => {
       // prevent downstream exporter calls from generating spans
       context.with(suppressInstrumentation(context.active()), () => {
-        this._exporter.export(this._finishedSpans, result => {
-          this._finishedSpans = [];
+        const batch = this._finishedSpans;
+        // Reset the finished spans buffer here because the next invocations of the _flush method
+        // could pass the same finished spans to the exporter if the buffer is cleared
+        // outside of the execution of this callback.
+        this._finishedSpans = [];
+        this._exporter.export(batch, result => {
```

Comment:
this could be a one liner without line 112 and 116
```suggestion
        this._exporter.export(this._finishedSpans.splice(0), result => {
```